### PR TITLE
[Reviews] Adding templates for github and CONTRIBUTING.md

### LINKED ISSUE #12: Create Contributing Guidelines
`CONTRIBUTING.md` should provide an overview of (roughly):

- git cloning, branching, rebasing, pushing
- QA/shellcheck/CI and style
- Creating PRs

### PR DESCRIPTION
**Pull request by btamayo**
Wanted to open this PR since there's some items like the **Formatting** section and **Code Conventions** that we should probably look over.

@mbland  Thanks so much for letting us use your original CONTRIBUTING.md from https://github.com/mbland/go-script-bash. Please feel free to update as you see fit, remove/add sections, change wording, etc.

### [The **diff for the two documents** is here.](https://www.diffchecker.com/I70X2B3C)

I personally don't have much experience with golang. I trimmed the bits that pertained to go, but if it's applicable _please tell me_ and I'll add it back. :) I also tried to be slightly more brief to anticipate added sections later.

This closes #12.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,355 @@
...
+  the exit status will always indicate success, as it is the status of the
+  `local` declaration, not the command substitution.
+- To work around a bug in some versions of Bash whereby arrays declared with
+  `declare -g` or `export` and initialized in the same statement eventually go
+  out of scope, always `export` the array name on one line and initialize it the
+  next line. See:
+  - https://lists.gnu.org/archive/html/bug-bash/2012-06/msg00068.html
+  - ftp://ftp.gnu.org/gnu/bash/bash-4.2-patches/bash42-025
+  - http://lists.gnu.org/archive/html/help-bash/2012-03/msg00078.html
+
```
Comment (sublimino) [inline CONTRIBUTING.md]: could add ShellCheck hint? 

```
- [ShellCheck](https://www.shellcheck.net/) can help to identify many of these issues 
```

**Comment by sublimino** [comment]
@mbland's style guide looks like a great place to start. 

I've PR'd some minor tweaks to @btamayo's branch, and there's a question above of whether we like ShellCheck enough to add a link to it (there are lots of plugins for IDEs, I think it's marvellous). Apart from that it's all good AFAICS.

**Comment by sublimino** [comment]
Pending a merge of minor changes to the source branch of this PR it's good to merge IMO. 

And in the interest of https://github.com/bats-core/bats-core/milestone/1 I'm inclined to merge it this week regardless unless @mbland or @jasonkarns feel otherwise? :smile:

**Comment by jasonkarns** [comment]
no objection

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,355 @@
...
+- memory usage as reported by a tool such as
+  [memusg](https://gist.github.com/netj/526585)
+
+### On existing issues
+
+1. DO NOT add a +1 comment: Use the reactions provided instead
+2. DO add information if you're facing a similiar issue as someone else, but 
+within a different context (e.g. different steps needed to reproduce the issue 
+than previous stated, different version of bash or bats, different OS, etc.) 
+You can read on how to do that here: [Information to include][#information-to-include]
```
Comment (mbland) [inline CONTRIBUTING.md]: The "Information to include" section is missing. Did you mean to add it?

File: CONTRIBUTING.md (hunk 1 above)
Comment (btamayo) [reply CONTRIBUTING.md]: Done! Good idea :)

File: CONTRIBUTING.md (hunk 2 above)
Comment (btamayo) [reply CONTRIBUTING.md]: PR Updated :)

**Pull request merged**